### DISCUSSION TITLE
Restore from LineageOS 19.1 to LineageOS 16

### DISCUSSION THREAD

**Discussion by galantra** [seedvault-app/seedvault#417] [Q&A] [answered]
I used to use LineageOS 19.1 on a OnePlus 6 and I made cloud backups with seedvault. Then my phone broke (CrashDump error); it is effectively bricked now. 

So, I set up [LineageOS 16 on Huawei P9; a higher LineageOS version is not available](https://forum.xda-developers.com/t/rom-treble-microg-lineageos-16-0-for-huawei-p9.4078383/).

Is there any way how I can restore my Seedvault backups (which I still have in .SeedVaultAndroidBackup in cloud storage) to LineageOS 16?

[Here](https://github.com/yijuchung/Neo-Backup/blob/main/FAQ.md) it is said that Seedvault relies on Google's api to backup… which gives me some hope.

**Comment by galantra** [comment]
Thanks to https://www.xda-developers.com/lineageos-seedvault-open-source-backup-solution/ I know now that I should be able to use https://github.com/tlambertz/seedvault_backup_parser

Unfortunately, I get an AssertionError – like two other users (according to the github issues).

**Comment by chirayudesai** [comment, marked as answer]
Unfortunately downgrades are not supported, due to how data is stored and handled - newer versions often change how data is stored, and upgrades to that are usually one way.

In theory app data could be restored but that's not something we have implemented at the moment.

Also SeedVault wasn't supported on that old of an Android either, so even if downgrades worked in general they still wouldn't help you.
